test/ostruct/test_ostruct.rb

```diff
@@ -100,6 +100,16 @@ def s.to_sym
     o.delete_field s
     assert_not_respond_to(o, :foo)
     assert_not_respond_to(o, :foo=)
+
+    assert_raise(NameError) { o.delete_field(s) }
+    assert_equal(s.to_sym, o.delete_field(s) { :foo })
```

Comment:
This test is confusing because of the coincidence that the block returns the same things as `s.to_sym` and even more so because the first argument is `s.to_sym` when it should really be `:foo`.
Please change to `assert_equal(:bar, o.delete_field(s) { :bar })`

Reply:
Test upated